src/models/mod.rs

```diff
@@ -124,10 +124,7 @@ pub async fn select_retry_or_panic(
 pub(crate) async fn start_after_interruption(
     pool: &sqlx::Pool<sqlx::Postgres>,
 ) -> anyhow::Result<u64> {
-    let query = "SELECT block_height
-                        FROM near_balance_events
-                        ORDER BY block_timestamp desc
-                        LIMIT 1";
+    let query = "SELECT max(block_height) FROM near_balance_events";
```

Comment:
Good!
In the initial query, I wanted to say `ORDER BY event_index desc`, it runs fast. I sorted by timestamp automatically, my fault
But anyway, I like your proposed solution more